### DISCUSSION TITLE
Deviation between the reconstructed PF and the experimental PF

### DISCUSSION THREAD

**Discussion by PSBHS** [ORTools4MTEX/ORTools#46] [Q&A] [answered]
Hello Frank and Azdiar, 

When I checked the pole figure of one reconstructed parent austenite with my experimental result, I found the deviation was relatively large, as shown below. Similar deviation can also be found in the PF of child martensite. Could you give some suggestions to fix the problem?

![image](https://github.com/ORTools4MTEX/ORTools/assets/138580189/6b45946c-0c02-4473-80dc-868135ae7234)

Best

**Comment by AzdiarGazder** [comment]
Hi @PSBHS,

Without your script and map data, it is a little difficult to pinpoint the exact reason behind the deviation.
We've provided examples on how to go about reconstruction - but the ORs and values that enable it are specific to the ebsd map data being used.
So I suggest having a detailed look at optimising your reconstruction to suit your ebsd data first.
Also, please refer to the examples to see if you have missed any steps along the way.

Hope this helps.

Warm regards,
Azdi

**Reply by PSBHS** [reply]
Hello Azdiar,

I have tried the code of examples 1 and 7. But the deviation didn't change too much. I changed the initial guess of OR from KS to NW but the reconstructed result remained unchanged. I provide detailed information on the calculated OR, as shown below. The probabilities are very low. The documentation on the MTEX website shows the probabilities are close to 100%. I checked the video of example 7 and I found the probabilities were also close to 0%. Maybe this is the reason for the large deviation? Hope for your answer!

![p2c](https://github.com/ORTools4MTEX/ORTools/assets/138580189/a09a0060-17e1-4413-ab6b-0e4d54585ee6)
![c2c](https://github.com/ORTools4MTEX/ORTools/assets/138580189/ae3e00f6-0816-44c0-9176-049500632be2)
![ORinfo](https://github.com/ORTools4MTEX/ORTools/assets/138580189/5d85153c-e309-4e23-b19b-8907a2de1d12)
![probainfo](https://github.com/ORTools4MTEX/ORTools/assets/138580189/7d39819f-5bc7-40f0-b1e0-e192b776ca4f)

Best wishes

**Reply by PSBHS** [reply]
Hello Azdiar,

I misunderstand the probability mentioned above. I have understood it now. I tried to change the threshold or tolerance that could be modified in the code but the deviation still existed without any change. I am really confused about the result. The following link contains my EBSD data and script. Could you please help to check the code? Hope for your answer.

[https://connecthkuhk-my.sharepoint.com/:f:/g/personal/pans_connect_hku_hk/EqM2Eq8pEoxCgW7wGE2VqfwB2zQBebnWFTkcZxVfp1fyfg?e=tHkdNI](url)

Best regards

**Comment by PSBHS** [comment]
Hello Azdiar,

I cropped the raw data of the corresponding parent austenite grain. Then I import the date to the MTEX and plot the pole figure directly. I still observe the deviation.  Why?
![image](https://github.com/ORTools4MTEX/ORTools/assets/138580189/c82ef31a-6b71-4485-9c24-9371816ac052)

Best regards

**Reply by AzdiarGazder** [reply]
Hi @PSBHS,

@frankNiessen and I are having a look at your data.
Please give us some time to understand all issues.

Warm regards,
Azdi

**Reply by AzdiarGazder** [reply]
Hi @PSBHS ,

A couple of issues:

1. I noticed an issue with the way you are loading your ang file.
```matlab
ebsd = EBSD.load(fname,CS,'interface','ang',...
  'convertEuler2SpatialReferenceFrame');
```
which, given that you are using OIM should be something like this:
```matlab
ebsd = EBSD.load([Ini.ebsdPath,'900-10min-77K-90min.ang'],'interface','ang','setting 2','convertEuler2SpatialReferenceFrame');
```
Please re-check the MTEX documentation for the correct setting (which corresponds to your microscope setup) and the correct reference frame as well.

2. I'm having trouble replicating your deviation error.
Could you please confirm whether or not you are running the developer version of MTEX and ORTools?
Just as a precaution, could you please re-install the developer versions of MTEX and ORTools and check if the error persists after you have addressed point 1 (above)?

Warm regards,
Azdi

**Reply by AzdiarGazder** [reply]
Hi @PSBHS,

I have tried to replicate your deviation issue but unfortunately I do not get the error at all.
I'd like you to try the following:

1. Please download and re-install the latest developer versions of MTEX and ORTools. Please download from today as some changes were implemented overnight.
2. Please copy the EBSD map to the "...Documents\MATLAB\GitHub\ORTools\data\input\ebsd" directory.
3. Copy and paste the attached script to "...\Documents\MATLAB\GitHub\ORTools" directory.
4. Run the script. Please remember that there are 2 "return" commands in the script that force you to run the script in separate blocks. 
5. Use the peak fitter **for the largest peak ONLY** to find your optimal OR. This may take some time - but please be patient. Here you will notice that your optimal OR is in-between NW and GT.
6. At the end, run the grainClick command. An IPF of child orientations has been implemented. At this point I do not see any deviation whatsoever.

Hope this helps.

Warm regards,
Azdi

```matlab

% Reconstructing the prior austenite microstructure from lath martensite
% with the variant graph approach and analyzing variant pairing
% The MTEX description for reconstruction is given here:
% https://mtex-toolbox.github.io/MaParentGrainReconstruction.html
% *********************************************************************
home; close all; clear variables;
currentFolder;
set(0,'DefaultFigureWindowStyle','normal');
screenPrint('StartUp','ORTools - 900-10min-77K-90min.ang');

%% Initialize MTEX
% Startup and set some settings
startup_mtex;
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','outOfPlane');
setMTEXpref('FontSize',14);   
setInterp2Tex;

% Default directories - Do not modify
Ini.dataPath = [strrep(pwd,'\','/'),'/data/'];
Ini.cifPath = [Ini.dataPath,'input/cif/'];
Ini.ebsdPath = [Ini.dataPath,'input/ebsd/'];
Ini.texturePath = [Ini.dataPath,'output/texture/'];
Ini.imagePath = [Ini.dataPath,'output/images/'];

%% Load data
% Load an MTEX dataset into 'ebsd'
% create an EBSD variable containing the data
ebsd = EBSD.load([Ini.ebsdPath,'900-10min-77K-90min.ang'],'interface','ang','setting 2','convertEuler2SpatialReferenceFrame');

%% Compute, filter and smooth grains
screenPrint('SegmentStart','Computing, filtering and smoothing grains');
% Grains are calculated with a 3° threshold
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',3*degree);
% EBSD data in small grains are removed
ebsd(grains(grains.grainSize < 10)) = [];
% Recalculate the grains from the remaining data ...
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',3*degree);
% ... and smooth the grain boundaries
grains = smooth(grains,5);

%% Rename and recolor phases 
screenPrint('SegmentStart','Renaming and recoloring phases');
phaseNames = {'AlphaP','Gamma'};
% Rename 'Fe (alpha, ferrite, bcc)' to 'AlphaP' and 'Fe (gamma, austenite, fcc)' to 'Gamma'
ebsd = renamePhases(ebsd,phaseNames);
% Choose your favourite colors
ebsd = recolorPhases(ebsd);

%% Define and refine parent-to-child orientation relationship
screenPrint('SegmentStart','Define and refine parent-to-child OR');
% Define 'Gamma' as the parent and 'AlphaP' as the child phase
job = setParentGrainReconstructor(ebsd,grains,Ini.cifPath);
% Find the optimal OR via peak-fitting...
job = defineORs(job);
% ... and refine it based on the fit with boundary misorientations
job.calcParent2Child;
% Display information about the OR
ORinfo(job.p2c);


% Compare the optimal OR with all other known ORs 
% (This is done by checking the disorientation or misfit between the 
% grain misorientations and an OR's theoretical misorientation)

%--- Define the theoretical planes and directions of the Bain OR
% fcc theoretical plane and direction
hklP = Miller(0,1,0,job.csParent); uvwP = Miller(0,0,1,job.csParent);
% bcc theoretical plane and direction
hklC = Miller(0,1,0,job.csChild); uvwC = Miller(1,0,1,job.csChild);
B = orientation.map(hklP,hklC,uvwP,uvwC);
%---

%--- Define the theoretical planes and directions of the Kurdjumov-Sachs OR
% % REF: https://doi.org/10.1016/j.actamat.2014.03.059
% fcc theoretical plane and direction
hklP = Miller(1,1,1,job.csParent); uvwP = Miller(-1,0,1,job.csParent);
% bcc theoretical plane and direction
hklC = Miller(0,1,1,job.csChild); uvwC = Miller(-1,-1,1,job.csChild);
KS = orientation.map(hklP,hklC,uvwP,uvwC);
%---

%--- Define the theoretical planes and directions of the Nishiyama-Wassermann OR
% % REF: https://doi.org/10.1016/j.actamat.2014.03.059
% fcc theoretical plane and direction
hklP = Miller(1,1,1,job.csParent); uvwP = Miller(1,1,-2,job.csParent);
% bcc theoretical plane and direction
hklC = Miller(0,1,1,job.csChild); uvwC = Miller(0,-1,1,job.csChild);
NW = orientation.map(hklP,hklC,uvwP,uvwC);
%---

%--- Define the theoretical planes and directions of the Pitsch OR
% (also known as inverse Nishiyama-Wassermann)
% % REF: https://doi.org/10.1016/j.actamat.2014.03.059
% fcc theoretical plane and direction
hklP = Miller(0,1,0,job.csParent); uvwP = Miller(-1,0,1,job.csParent);
% bcc theoretical plane and direction
hklC = Miller(1,-1,0,job.csChild); uvwC = Miller(-1,-1,1,job.csChild);
P = orientation.map(hklP,hklC,uvwP,uvwC);
%---

%--- Define the theoretical planes and directions of the Greninger-Troiano OR
% % REF: https://doi.org/10.1016/j.actamat.2014.03.059
% fcc theoretical plane and direction
hklP = Miller(1,1,1,job.csParent); uvwP = Miller(5,12,-17,job.csParent);
% bcc theoretical plane and direction
hklC = Miller(0,1,1,job.csChild); uvwC = Miller(7,-17,17,job.csChild);
GT = orientation.map(hklP,hklC,uvwP,uvwC);
%---

%--- Define the theoretical planes and directions of the inverse Greninger-Troiano OR
% % REF: https://doi.org/10.1016/j.actamat.2014.03.059
% fcc theoretical plane and direction
hklP = Miller(17,7,17,job.csParent); uvwP = Miller(-1,0,1,job.csParent);
% bcc theoretical plane and direction
hklC = Miller(5,12,17,job.csChild); uvwC = Miller(-1,-1,1,job.csChild);
iGT = orientation.map(hklP,hklC,uvwP,uvwC);
%---

%--- Define the theoretical planes and directions of the Headley-Brooks OR
% % REF: https://link.springer.com/article/10.1007/s11661-002-0001-0
% fcc theoretical plane and direction
hklP = Miller(1,1,1,job.csParent); uvwP = Miller(-1,1,0,job.csParent);
% bcc theoretical plane and direction
hklC = Miller(1,1,0,job.csChild); uvwC = Miller(-1,1,0,job.csChild);
HB = orientation.map(hklP,hklC,uvwP,uvwC);
%---

plotHist_OR_misfit(job,[B,KS,NW,P,GT,iGT,HB],'legend',{'B OR','K-S OR','N-W OR','iN-W(P) OR','G-T OR','iG-T OR','H-B OR'});

return
%% Plotting (with ORTools functions)
screenPrint('SegmentStart','Plotting some ORTools maps');
% Use some of the ORTools functions to visualize the determined OR
% and its relation to the microstructure

% Phase map
plotMap_phases(job,'linewidth',1);
%    - There is no retained austenite (gamma)

% Parent and child IPF maps
plotMap_IPF_p2c(job,vector3d.Z,'linewidth',1);
%    - It is clear that martensite has formed from multiple prior 
%    - austenite grains and that some surface scratches led to bad indexing

% Child-child grain boundary misorientation map
plotMap_gB_c2c(job,'linewidth',1);
%    - Misorientation angles of ~15-50° are not present within prior
%    austenite grains and thus delinitate prior austenite grain
%    boundaries

% Plot a map of the OR boundary disorientation, or misfit
plotMap_gB_misfit(job,'linewidth',1,'maxColor',5);
%    - By setting a threshold at 5 degrees, the prior austenite grain
%    boundaries are identified by their large misfit with the OR

% Plot a map of the OR boundary probability 
plotMap_gB_prob(job,'linewidth',1);
%   - the same can be visualized by calculating the probability that a
%     boundary belongs to the OR

%% Reconstruct parent microstructure
%   - Reconstruct the microstructure with the variant graph based approach
job.calcVariantGraph('threshold',5*degree,'tolerance',2.5*degree,'mergeSimilar')
job.clusterVariantGraph('includeSimilar');
% ... plot the votes (high values show high certainty)
figure; plot(job.grains,job.votes.prob(:,1))
mtexColorbar
% ... and calculate the parent orientations
job.calcParentFromVote('minProb',0)
% Plot the reconstructed parent microstructure
figure;
plot(job.parentGrains,job.parentGrains.meanOrientation,'linewidth',1);

%% Remove badly reconstructed clusters
% In order to reconstruct the remaining parent grains, we can calculate the
% votes for surrounding parent grains by the already reconstructed parent
% grains

% compute the votes
job.calcGBVotes('p2c','reconsiderAll')
% assign parent orientations according to the votes
job.calcParentFromVote
% plot the result
figure;
plot(job.parentGrains,job.parentGrains.meanOrientation)

%% Clean reconstructed grains
% Now clean the grains by: 
% - merging grains with similar orientation
job.mergeSimilar('threshold',7.5*degree);
% - merging small inclusions
job.mergeInclusions('maxSize',250);
% This is the cleaned reconstructed parent microstructure
figure;
plot(job.parentGrains,job.parentGrains.meanOrientation,'linewidth',1)

%% Variant analysis
% Calculate the variant IDs of all alpha grains
job.calcVariants;
% ... and plot them
plotMap_variants(job,'grains','linewidth',1); %Grain data
plotMap_variants(job,'linewidth',1); %EBSD data

% We see that each child grain contains several variants on EBSD level
% For analyzing variant pairing, we need the variants on the EBSD level
% to be reconstructed as grains
[newGrains,newEBD] = computeVariantGrains(job);
%Compare variant indexing for old and new grains
figure; %Old grains
plot(grains(job.csChild),grains(job.csChild).meanOrientation);
figure; %New grains
plot(newGrains(job.csChild),newGrains(job.csChild).meanOrientation);

return
%% Display variant information for a PAG by interactively clicking on a gamma grain
grainClick(job,'noScalebar','noFrame'); %Based on EBSD data

%% Do the same at the grain level 
grainClick(job,'grains','noFrame'); %Based on grain data

%% Refine gamma twins for a PAG by interactively clicking on a gamma grain
grainClick(job,'parentTwins');


```

**Reply by PSBHS** [reply]
Thank you, Azdiar! I'll try it.

**Reply by PSBHS** [reply]
Hello Azdiar,

I have tried the code you provided but I can still observe the deviation, as shown below. There is no problem with the code and no error was reported during the calculation. 
![幻灯片2](https://github.com/ORTools4MTEX/ORTools/assets/138580189/c42d1fae-b475-45a8-8b6f-93bd04518f5a)
Besides, I also run the example 1 and got the PF of one parent austenite grain. I cropped the same region and got the PF by Channel 5. These two PFs highly coincide with each other. I don't understand the reason. Hope for your answer!
![幻灯片1](https://github.com/ORTools4MTEX/ORTools/assets/138580189/3a184f62-b034-4847-84d6-38f3a6ff461c)

Best regards

**Reply by AzdiarGazder** [reply, marked as answer]
Hi @PSBHS,

Frank and I have access to and use the Oxford Instruments software suites to gather and/or analyse map data.
While we can load the *.ang files from OIM in MTEX, neither of us have access to the OIM software suite.
So, unfortunately, Frank and I are unable to replicate the deviation error you are getting.

The last suggestion I have is to check the default projection method/type being employed by MTEX and OIM in the pole figures (i.e. - equal area, equal angle, or equal distance).

Warm regards,
Azdi

**Reply by PSBHS** [reply]
Thank you! I'll check it.

**Reply by PSBHS** [reply]
Hello Azdiar,

You're right! I change the projection method in the OIM from stereographic to equal area and the two PFs finally coincide with each other! I find the definitions of different projection methods in the OIM software. Could these methods be added to the MTEX in the future? Thank you very much for your patient answer!
![幻灯片1](https://github.com/ORTools4MTEX/ORTools/assets/138580189/4a7785fd-0adc-4d70-810d-84c59f22d0fb)
![幻灯片2](https://github.com/ORTools4MTEX/ORTools/assets/138580189/ff44da76-0a83-45eb-9680-390a02953300)

Best regards

**Comment by AzdiarGazder** [comment]
Hi @PSBHS,

I'm glad that resolved the issue.

To address your comments, detailed explanations are provided in this [weblink](https://mtex-toolbox.github.io/SphericalProjections.html).
As you can see, these projections are already part of MTEX.

You can go into the various functions of ORTools and change how the spherical projection is displayed (default = equal area) by including the 'projection' option as shown in the example script given below.

```matlab
cs = crystalSymmetry('m-3m');
odf = fibreODF(Miller(1,1,0,cs),zvector)

figure % the equal area projection (Schmidt projection, MTEX default)
plotPDF(odf,Miller(1,0,0,cs),'antipodal')

figure % the equal area projection (Schmidt projection, MTEX default)
plotPDF(odf,Miller(1,0,0,cs),'projection','earea','antipodal')

figure % the equal angle projection (stereographic projection)
plotPDF(odf,Miller(1,0,0,cs),'projection','eangle','antipodal')

figure % the equal distance projection
plotPDF(odf,Miller(1,0,0,cs),'projection','edist','antipodal')
```

Warm regards,
Azdi

**Reply by PSBHS** [reply]
Ok, I'll try it. Thank you!
